terraform/aws/components/base/vpc_peering_dlt.tf

```diff
@@ -0,0 +1,76 @@
+# vpc peering to the network where DLT is hosted
+data "aws_vpc" "dlt_vpc" {
+  id = var.dlt_vpc_id
+}
+
+resource "aws_vpc_peering_connection" "dlt_peering" {
+  vpc_id = aws_vpc.base_vpc.id
+  peer_vpc_id = data.aws_vpc.dlt_vpc.id
+  auto_accept = true
+
+  accepter {
+    allow_remote_vpc_dns_resolution = true
+  }
+
+  requester {
+    allow_remote_vpc_dns_resolution = true
+  }
+
+  tags = merge(local.default_tags,{
+    Name = "${local.resource_prefix}-dlt-vpc-peering"
+  })
+}
+
+# Add a route to the NHAIS VPC in the DLT VPC route table
+resource "aws_route" "dlt_to_base_route" {
+  route_table_id = data.aws_vpc.dlt_vpc.main_route_table_id
+  destination_cidr_block = aws_vpc.base_vpc.cidr_block
+  vpc_peering_connection_id = aws_vpc_peering_connection.dlt_peering.id
+  depends_on = [aws_vpc_peering_connection.dlt_peering]
+}
+
+# Add route to the NHAIS VPC from the second DLT route table
+resource "aws_route" "nhais_to_dlt_route" {
+  route_table_id = var.second_dlt_route_id
+  destination_cidr_block = aws_vpc.base_vpc.cidr_block
+  vpc_peering_connection_id = aws_vpc_peering_connection.dlt_peering.id
+  depends_on = [aws_vpc_peering_connection.dlt_peering]
+}
+
+# Add a route to the DLT VPC in the NHAIS VPC route table
+resource "aws_route" "base_to_dlt_route" {
+  route_table_id = aws_route_table.private.id
+  destination_cidr_block = data.aws_vpc.dlt_vpc.cidr_block
+  vpc_peering_connection_id = aws_vpc_peering_connection.dlt_peering.id
+  depends_on = [aws_vpc_peering_connection.dlt_peering]
+}
+resource "aws_security_group_rule" "nhais_inbound_security_group_amazon_dlt_ingress_rule" {
```

Comment:
This rule is not needed, and does not meet the intended use for core SG.
Core SG is there to allow all resources in Env to certain things like VPC Endpoints, NAT, and in this case the DLT.
Ingress from DLT to env should be done on the SG for Load Balancer

Reply:
Cool, let me remove test and get back to you.

Reply:
Removed now

Reply:
Done,